pkg/apis/batch/validation/validation.go

```diff
@@ -698,7 +698,15 @@ func ValidateJobStatusUpdate(job, oldJob *batch.Job, opts JobStatusValidationOpt
 	if opts.RejectStartTimeUpdateForUnsuspendedJob {
 		// Note that we check `oldJob.Status.StartTime != nil` to allow transitioning from
 		// startTime = nil to startTime != nil for unsuspended jobs, which is a desired transition.
-		if oldJob.Status.StartTime != nil && !ptr.Equal(oldJob.Status.StartTime, job.Status.StartTime) && !ptr.Deref(job.Spec.Suspend, false) {
+
+		// Detect job resume via condition changes (JobSuspended: True -> False)
+		// This handles the case where the controller updates status after the user has already
+		// changed spec.suspend=false, which is the scenario from https://github.com/kubernetes/kubernetes/issues/134521
+		isJobResuming := IsConditionTrue(oldJob.Status.Conditions, batch.JobSuspended) &&
+			!IsConditionTrue(job.Status.Conditions, batch.JobSuspended)
```

Comment:
I'm thinking it will be cleaner to move this condition under computation of RejectStartTimeUpdateForUnsuspendedJob, because:
1.the name RejectStartTimeUpdateForUnsuspendedJob wouldn't capture the extra check currently
2. it will be clear from the context that the fix is behind the JobManagedBy feature gate, now it feels disconnected